include/wobbly/geometry.h

```diff
@@ -0,0 +1,352 @@
+            };
+        }
+
+        /**
+         * PointView<NumericType>:
+         *
+         * A view of a single point into an array of two dimensional
+         * NumericTypes.
+         *
+         * This allows us to use "structure of arrays" and exploit cache
+         * locality when updating lots of forces, velocities or positions
+         * at the same time.
+         *
+         * PointView implements the Dimension trait, meaning that it
+         * can be used with functions in the wobbly::geometry::dimension
+         * namespace.
+         */ 
+        template <typename NumericType>
+        class PointView :
+            public detail::PointViewCopyabilityBase <NumericType>::type
+        {
+            public:
+
+                typedef NumericType NT;
+                typedef typename std::remove_const <NumericType>::type NTM;
+                typedef typename std::add_const <NumericType>::type NTC;
+
+                typedef typename std::is_const <NT> IC;
+
+                template <std::size_t N>
+                PointView (std::array <NTM, N> &points,
+                           std::size_t         index) :
+                    array (points.data ()),
+                    offset (index * 2)
+                {
+                }
+
+                template <std::size_t N,
+                          typename = typename std::enable_if <IC::value && N>::type>
+                PointView (std::array <NTM, N> const &points,
+                           std::size_t               index) :
+                    array (points.data ()),
+                    offset (index * 2)
+                {
+                }
+
+                PointView (NumericType *points,
+                           std::size_t index) :
+                    array (points),
+                    offset (index * 2)
+                {
+                }
+
+                PointView (PointView <NumericType> &&view) :
+                    array (view.array),
+                    offset (view.offset)
+                {
+                }
+
+                PointView &
+                operator= (PointView <NumericType> &&view) noexcept
+                {
+                    if (this == &view)
+                        return *this;
+
+                    array = view.array;
+                    offset = view.offset;
+
+                    return *this;
+                }
+
+                /* Copy constructor and assignment operator enabled only if the view
+                 * is truly just observing and has no write access to the point,
+                 * see detail::PointViewCopyabilityBase */
+                PointView (PointView <NumericType> const &p) :
+                    array (p.array),
+                    offset (p.offset)
+                {
+                }
+
+                friend void swap (PointView <NumericType> &first,
+                                  PointView <NumericType> &second)
+                {
+                    using std::swap;
```

Comment:
Nitpick: I'd prefer to just use `std::swap` in the calls below. Makes it clearer.

Reply:
Makes sense